src/components/wrapper/song-list-wrapper.tsx

```diff
@@ -0,0 +1,158 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react"
+import { PlusIcon } from "@heroicons/react/24/solid"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { Suggestion } from "@/types/database-types"
+import { MagnifyingGlassCircleIcon } from "@heroicons/react/24/outline"
+import Spinner from "@/components/utils/spinner"
+import ErrorPopup from "@/components/popups/error-popup"
+import { useRouter } from "next/router"
+import { MagnifyingGlassIcon } from "@heroicons/react/20/solid"
+import SearchBar from "@/components/suggestion/search-bar"
+import { getRepertoireSongs, getSuggestions } from "@/services/suggestion.service"
+
+interface SongListPageWrapperProps {
+  renderSongCard: (suggestion: Suggestion) => JSX.Element
+  pageName: string
+}
+
+const SongListPageWrapper = (props: SongListPageWrapperProps) => {
+  const router = useRouter()
+  const supabaseClient = useSupabaseClient<Database>()
+  const [showSearchBar, setShowSearchBar] = useState<boolean>(false)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const [songs, setSongs] = useState<Suggestion[]>([])
+  const [showSpinner, setShowSpinner] = useState(true)
+  const [showLoadingError, setShowLoadingError] = useState(false)
+  const [searchedSong, setSearchedSong] = useState<Suggestion[]>([])
+  const [errorText, setErrorText] = useState("")
+
+  useEffect(() => {
+    setShowSpinner(true)
+    const songs =
+      props.pageName == "Suggestions"
+        ? getSuggestions(supabaseClient)
+        : getRepertoireSongs(supabaseClient)
+
+    songs
+      .then((response) => {
+        if (response.error) {
+          setShowLoadingError(true)
+          return
+        }
+
+        if (response.data?.length > 0) {
+          setSongs(response.data as Suggestion[])
+          setErrorText("")
+        } else {
+          setErrorText(
+            `Looks like there are no ${
+              props.pageName === "Suggestions" ? "Suggestions made" : "Songs in repertoire"
+            } yet! Feel free to start adding them.`
```

Comment:
Users (except for admins) can't add songs to the repertoire so don't show that to them.
Also looks weird that Suggestions and Song start with a capitalized letter
![image](https://github.com/ASVGay/the-rhapsodies/assets/54884540/66a7da70-fae3-4e18-aaad-fa843b002fb7)